### PR TITLE
Add CI for installed CMake

### PR DESCRIPTION
**Pull request by vrabaud**
This PR was originally a backport of https://github.com/microsoft/vcpkg/pull/30824 but it is not needed anymore
It is now only about adding a CI for installed CMake file and removing the deprecated CMAKE_THREAD_PREFER_PTHREADS.

### CODE REVIEW

**Review by y-guyon — APPROVED** [review]
A little more explanation of the added steps in CMakeLists.txt and why they are needed would be appreciated (as comments in CMakeLists.txt for example).

File: .github/workflows/ci-unix-shared-installed.yml [hunk 1]
```diff
@@ -70,10 +70,29 @@ jobs:
         -DAVIF_CODEC_DAV1D=ON
         -DAVIF_BUILD_EXAMPLES=ON -DAVIF_BUILD_APPS=ON
         -DAVIF_BUILD_TESTS=ON -DAVIF_ENABLE_GTEST=ON -DAVIF_LOCAL_GTEST=ON
-        -DAVIF_BUILD_GDK_PIXBUF=ON
+        -DAVIF_BUILD_GDK_PIXBUF=ON -DCMAKE_INSTALL_PREFIX=./install/
```
Comment (wantehchang) [inline .github/workflows/ci-unix-shared-installed.yml:73]: Nit: `CMAKE_INSTALL_PREFIX` does not need to end with '/', right?

**Review by wantehchang — COMMENTED** [review]
Vincent: Please consider my comments as supplemental (i.e., a "drive-by" review).

File: CMakeLists.txt [hunk 2]
```diff
@@ -765,6 +765,31 @@ if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
         install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config-version.cmake
                 DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
         )
+        file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake.in "@PACKAGE_INIT@\n")
+        if(UNIX AND NOT BUILD_SHARED_LIBS)
```
Comment (kmilos) [inline CMakeLists.txt:769]: Threads are also available on MINGW, should it be included here as well?

Comment (vrabaud) [reply CMakeLists.txt]: Done

File: .github/workflows/ci-unix-shared-installed.yml [hunk 3]
```diff
@@ -70,10 +70,29 @@ jobs:
...
+              project(FindPackageTest)
+              message(STATUS ${CMAKE_PREFIX_PATH})
+              find_package(libavif REQUIRED)
+              if(TARGET avif)
+                message(STATUS "\""avif found"\"")
+              else()
+                message(FATAL_ERROR "\""avif not found"\"")
+              endif()" >> CMakeLists.txt
+
+        cmake ./ -DCMAKE_PREFIX_PATH=../install/
```
[the remark this answers is not in the document]
Comment (vrabaud) [reply .github/workflows/ci-unix-shared-installed.yml]: Done

File: .github/workflows/ci-unix-shared-installed.yml (hunk 1 above)
Comment (vrabaud) [reply .github/workflows/ci-unix-shared-installed.yml]: Done

**Comment by vrabaud** [comment]
The PR is now much simpler, I should have checked for the presence of `Threads` in the .cmake file.

File: CMakeLists.txt [hunk 4]
```diff
@@ -339,7 +339,6 @@ set(CMAKE_FIND_FRAMEWORK LAST)
 
 if(UNIX)
```
[the remark this answers is not in the document]
Comment (kmilos) [reply CMakeLists.txt:340]: Yep, native threading model seems to be used:

https://github.com/AOMediaCodec/libavif/blob/df7a543f2f9c750b11e4a3ea3cee0c6f1303b5bb/src/reformat.c#L10

 so no need, sorry for the side-track.

[the remark this answers is not in the document]
Comment (vrabaud) [reply CMakeLists.txt:340]: Right. Still, all that could be cleaned up, cf https://github.com/AOMediaCodec/libavif/pull/1608

**Review by y-guyon — APPROVED** [review]

**Pull request merged**